napari/_vispy/visuals/clipping_planes_mixin.py

```diff
@@ -0,0 +1,25 @@
+from ..vendored.filters.clipping_planes import PlanesClipper
+
+
+class ClippingPlanesMixin:
+    """
+    Mixin class that attaches clipping planes filters to the (sub)visuals
+    and provides property getter and setter
+    """
+
+    def __init__(self, *args, **kwargs):
+        self._clip_filter = PlanesClipper()
+        super().__init__(*args, **kwargs)
+
+        # get subvisuals if compound, or just self
```

Comment:
I might be using the terminology completely wrong, so please correct me if "mixin" is not the right therm here :)
The reason this exists is mainly because `Volume` behaves differently from all the other visuals (no use of the filter, despite having the same high level api of `Volume.clipping_planes = value`), so plain old inheritance was not the right way. do you have a better suggestion?

Reply:
I'm not familiar enough to give strong guidance on whether mixin is the right term here - but from I what I understand of mixins and your intent, I think it is because as you said you want to some behavior to both Mesh and Compound visuals, but not Volume visuals.

Your recent update is an improvement because now the assumption about the class that uses the mixin is simpler - that it implements an `attach` method.

If it's well understood that `ClippingPlanesMixin` will always be used by a class that implements or inherits an `attach` method, then this is good enough.

If you wanted to protect misuse of `ClippingPlanesMixin` (i.e. by classes that don't have an `attach` method), then you could move the call to attach in each of the Visuals' (e.g. `ShapesVisual.__init__`) - it's slightly more code (one line per Visual class), but it's also easier to understand because that line of code should be close to the declaration of the class where we can see it inherits from `Compound` or `Mesh` (i.e. the implementors of `attach`).

Reply:
if you like, you can indicate the intent of the Mixin (and provide some IDE support if someone uses it wrong) with:
```python
from typing import List, Optional
from typing_extensions import Protocol

class _PVisual(Protocol):
    _subvisuals: Optional[List['_PVisual']]
    def attach(self, filt: Filter, view=None ): ...

class ClippingPlanesMixin:
    """
    Mixin class that attaches clipping planes filters to the (sub)visuals
    and provides property getter and setter
    """

    def __init__(self: _PVisual, *args, **kwargs):
```